### PR TITLE
Lesson 8

### PR DESCRIPTION
**Pull request by IharDorinel**

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -487,3 +487,88 @@ function deepEqual (obj1, obj2) {
...
+
+class Student extends User {
+    constructor(firstName, lastName, admissionYear, courseName) {
+        super(firstName, lastName);
+        this.admissionYear = admissionYear;
+        this.courseName = courseName;
+    }
+    get course() {
+        return 2021 - `${this.admissionYear}`;
+    }
```
Comment (nikitik-nikita) [inline main.js]: let's change the receipt of the current year to a function:
`cosnt currentYear = new Date().getFullYear();`

File: main.js [hunk 2]
```diff
@@ -487,3 +487,88 @@ function deepEqual (obj1, obj2) {
...
+        this.admissionYear = admissionYear;
+        this.courseName = courseName;
+    }
+    get fullName1() {
+        return `${this.firstName} ${this.lastName}`
+    }
+    get course() {
+        return 2021 - `${this.admissionYear}`;
+    }
+}
```
Comment (nikitik-nikita) [inline main.js]: This is an unnecessary class

File: main.js [hunk 3]
```diff
@@ -487,3 +487,88 @@ function deepEqual (obj1, obj2) {
...
+
+const data2 = studentsData.map((e) => new Student1(...Object.values(e)))
```
Comment (nikitik-nikita) [inline main.js]: you can extands from `Student`

File: main.js [hunk 4]
```diff
@@ -487,3 +487,88 @@ function deepEqual (obj1, obj2) {
...
+
+class Students {
+    constructor(students) {
+        this.students = students;
+    }
+
+    getInfo() {
+        let sorted = this.students.sort((a, b) => a.course - b.course);
+        return sorted.map((a) => `${a.fullName1} - ${a.courseName}, ${a.course} курс`
```
Comment (nikitik-nikita) [inline main.js]: also need to change some values

**Review by nikitik-nikita — COMMENTED** [review]
Pls delete the same `data` array

**Comment by IharDorinel** [comment]
@nikitik-nikita I've fixed your comments

**Comment by nikitik-nikita** [comment]
@IharDorinel we still have this error in the console
errorPic

**Comment by IharDorinel** [comment]
@nikitik-nikita I've fixed it

**Pull request merged**